src/limbo/stat/gp.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef LIMBO_STAT_GP_HPP
+#define LIMBO_STAT_GP_HPP
+
+#include <cmath>
+
+#include <limbo/stat/stat_base.hpp>
+
+namespace limbo {
+    namespace stat {
+        /// @ingroup stat
+        /// filename: `gp_<iteration>.dat`
+        template <typename Params>
+        struct GP : public StatBase<Params> {
+            template <typename BO, typename AggregatorFunction>
+            void operator()(const BO& bo, const AggregatorFunction& afun, bool blacklisted)
+            {
+                if (!bo.stats_enabled())
+                    return;
+                std::string  fname = bo.res_dir() + "/" +
+                  "gp_" + std::to_string(bo.total_iterations()) + ".dat";
+                std::ofstream ofs(fname.c_str());
```

Comment:
I think we should write at the top of the file, the legend:

- what is being written (point, value, sigma, etc)
- number of dimensions of input of GP and number of dim for the output